lib/rdf/repository.rb

```diff
@@ -285,14 +304,10 @@ def has_statement?(statement)
       # @see RDF::Enumerable#each_statement
       def each_statement(&block)
         if block_given?
-          # Note that to iterate in a more consistent fashion despite
-          # possible concurrent mutations to `@data`, we use `#dup` to make
-          # shallow copies of the nested hashes before beginning the
-          # iteration over their keys and values.
-          @data.dup.each do |g, ss|
-            ss.dup.each do |s, ps|
-              ps.dup.each do |p, os|
-                os.dup.each do |o|
+          @data.each do |g, ss|
+            ss.each do |s, ps|
+              ps.each do |p, os|
+                os.each do |o|
                   # FIXME: yield has better performance, but broken in MRI 2.2: See https://bugs.ruby-lang.org/issues/11451.
                   block.call(RDF::Statement.new(s, p, o, graph_name: g.equal?(DEFAULT_GRAPH) ? nil : g))
```

Comment:
Yes, I think we can revert this. I saw it the other day too. I don't think we need to make any explicit statement about earlier patch revisions of MRI to be unsupported.